runtime/tr.source/trj9/ilgen/Walker.cpp

```diff
@@ -2186,7 +2186,7 @@ TR_J9ByteCodeIlGenerator::genTreeTop(TR::Node * n)
             // stashed
             //
             _couldOSRAtNextBC = true;
-            if ((n->getOpCode().isCheck() || n->getOpCodeValue() == TR::treetop) && n->getFirstChild()->getOpCode().isCall())
+            if ((n->getOpCode().isCheck() || n->getOpCodeValue() == TR::treetop) && n->getFirstChild()->getOpCode().isCall() && n->getFirstChild()->getReferenceCount() == 1)
```

Comment:
There's a lot of complexity in this code.  Is it worthwhile to document why the `referenceCount() == 1` check is required here in a comment?

Also the `n->getFirstChild()->getOpCode().isCall()` check must already be `true` due to the if statement on line 2177 that includes this exact condition.  Can this if condition be simplified?

Reply:
I added a comment to specify we are only interested in the first reference to the call.

I agree, this section of code has grow to be quite complex recently, as it must identify and prepare for all possible OSR points, but the `n->getFirstChild()->getOpCode().isCall()` won't always be `true`. Other types of OpCodes are managed by this section of code.

Reply:
You're right.  I misread the code above